### PR TITLE
Improve GradleProjectProperties.isWarProject() implementation

### LINKED ISSUE #1788: GradleProjectProperties.isWarProject() could be simpler
In its current implementation `com.google.cloud.tools.jib.gradle.GradleProjectProperties#isWarProject` will try to resolve the `war` task just to check if its null.

This might cause the `war` task to be created.
See https://docs.gradle.org/current/userguide/task_configuration_avoidance.html for more information why the unneccessary creation of tasks is discuraged in newer gradle versions.

The method could just check if the `war` plugin was applied to the gradle project.

### PR DESCRIPTION
**Pull request by larsgrefer**
fixes #1788

### CODE REVIEW

File: jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleProjectPropertiesTest.java [hunk 1]
```diff
@@ -59,9 +59,7 @@
 import org.gradle.api.java.archives.internal.DefaultManifest;
 import org.gradle.api.logging.Logger;
 import org.gradle.api.logging.configuration.ConsoleOutput;
-import org.gradle.api.plugins.Convention;
-import org.gradle.api.plugins.JavaPluginConvention;
-import org.gradle.api.plugins.WarPluginConvention;
+import org.gradle.api.plugins.*;
```
Comment (chanseokoh) [inline jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleProjectPropertiesTest.java]: Formatting problem. Running `./build.sh format` at the repo root should auto-format all code.

File: jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java [hunk 2]
```diff
@@ -278,7 +279,7 @@ public String getJarPluginName() {
 
   @Override
   public boolean isWarProject() {
-    return TaskCommon.getWarTask(project) != null;
+    return project.getPlugins().hasPlugin(WarPlugin.class);
```
Comment (chanseokoh) [inline jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java]: I'm guessing yes, but will this work with bootWar too?

Comment (larsgrefer) [reply jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java]: Yes. In fact the `bootWar` task is only created by the spring boot gradle plugin, if the `WarPlugin` is applied to the project.

File: jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleProjectPropertiesTest.java (hunk 1 above)
Comment (larsgrefer) [reply jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleProjectPropertiesTest.java]: `./build.sh format` doesn't change anything.

File: jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java (hunk 2 above)
Comment (chanseokoh) [reply jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java]: Hmm. Sorry, could you expand the import?

File: jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleProjectPropertiesTest.java (hunk 1 above)
Comment (chanseokoh) [reply jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleProjectPropertiesTest.java]: Hmm. Sorry, could you expand the import?

Comment (loosebazooka) [reply jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleProjectPropertiesTest.java]: I think you'll have to manually expand this then?

**Pull request merged**